apps/juicebox_web/lib/web/channels/stream_channel.ex

```diff
@@ -2,35 +2,61 @@ defmodule JuiceboxWeb.StreamChannel do
   use Phoenix.Channel
 
   alias Phoenix.PubSub
+  alias JuiceboxWeb.Presence
   alias JuiceboxStream.Stream.Server, as: Stream
+  alias JuiceboxWeb.Reactions.Server, as: Reactions
 
   intercept ["queue.updated"]
 
   def join("stream:" <> stream_id, _params, socket) do
+    JuiceboxStream.Stream.Supervisor.start_stream(stream_id)
+    JuiceboxWeb.Reactions.Supervisor.start_reactions(stream_id)
+
     PubSub.subscribe(JuiceboxStream.PubSub, "juicebox:stream:server:" <> stream_id)
-    IO.inspect JuiceboxStream.Stream.Supervisor.start_stream(stream_id)
+
     {:ok, queue} = Stream.queue(stream_id)
 
     send(self, {:after_join, stream_id})
 
+    socket = socket
+    |> assign(:user_id, uuid())
+    |> assign(:stream_id, stream_id)
+
     {:ok, socket}
   end
 
   def handle_in("video.added", %{"stream_id" => stream_id, "video" => video} = _, socket) do
     {:ok, state} = Stream.add(stream_id, %{ video: (for {key, val} <- video, into: %{}, do: { String.to_atom(key), val}) } )
 
-    {:noreply, socket}
+    {:reply, :ok, socket}
+  end
+
+  def handle_in("reaction.sent", %{"stream_id" => stream_id, "video" => video}, socket) do
+    Reactions.put(stream_id, socket.assigns.user_id, video)
+
+    broadcast_reaction(socket, %{ video: video, user_id: socket.assigns.user_id })
```

Comment:
There is no whitespace around parentheses/brackets most of the time, but here there is.